app/models/lecture.rb

```diff
@@ -1,3 +1,16 @@
 class Lecture < ApplicationRecord
   belongs_to :question_set
+  
+  #compute out of all the responses, how many corresponded to each answer
+  #returns hash keyed by answer id with value for % responses with that answer id
+  def self.compute_percentages (q)
+    @responses = Response.where(question: q)
+    @answers = Answer.where(question: q)
+    @total = @responses.length
+    hash = Hash.new(0)
```

Comment:
Ok that makes sense. But then I think you just want `Hash.new()` because otherwise I could index into the hash with anything and I'll at least get 0. If you omit the default 0, you'll still have 0 for any answer without responses, but `nil` for anything not recorded in a response